### DISCUSSION TITLE
rpc method with custom schema

### DISCUSSION THREAD

**Discussion by megacherry** [supabase/supabase#12689] [Questions] [answered]
Hi,

is it possible to extend the rpc-method to call a function that does not have the public schema?

best regards
Andy

**Comment by GaryAustin1** [comment, marked as answer]
Yes.  You have to allow it in API settings.
Then you can have a client go to an alternate schema when you initialize it.
![image](https://user-images.githubusercontent.com/54564956/221419250-990951e4-5e67-49b1-838c-88f4d2cd0a97.png)

Note you can have multiple clients, just set 
```
    autoRefreshToken: false,
    persistSession: false,
    detectSessionInUrl: false
```

**Reply by megacherry** [reply]
Thank you! I read about it long time ago. (: That means I have to create a separate connection for each scheme?

**Reply by GaryAustin1** [reply]
Yes, currently the only built in way is to init a client.
There are no connections though in Supabase.  It is a REST API.  So you could have 10 clients initialized and there is not network/database overhead until they generate a REST request.

**Reply by megacherry** [reply]
Perfect, thank you for the fast and accurate help.
